src/BaseFactory.php

```diff
@@ -46,16 +45,29 @@ protected function build(array $extra = [], string $creationType = 'create')
             return $model;
         }
 
-        $relatedModels = $this->relatedModelFactories->map->make();
-
-        if ($creationType === 'create') {
-            $model->{$this->relatedModelRelationshipName}()
-                    ->saveMany($relatedModels);
-
-            return $model;
+        foreach ($this->relatedModelFactories as $relationshipName => $factories) {
+            $relation = $model->{$relationshipName}();
+
+            if (method_exists($relation, 'saveMany')) {
```

Comment:
I think we can improve on this code section. Maybe with two "ifs" and early model returns and at the end throw the exception if the model was not returned? Maybe also @nuernbergerA has an idea?

Reply:
Maybe something like this?
```php
foreach ($this->relatedModelFactories as $relationshipName => $factories) {
    $relation = $model->{$relationshipName}();

    if (method_exists($relation, 'saveMany')) {
        $relatedModels = $factories->map->make();
        $model->setRelation($relationshipName, $relatedModels);

        if ($creationType === 'create') {
            $relation->saveMany($relatedModels);
        }

        continue;
    }

    if (method_exists($relation, 'associate')) {
        $relatedModels = $factories->map->create();
        $relatedModels->each(fn($related) => $relation->associate($related));

        if ($creationType === 'create') {
            $model->save();
        }

        continue;
    }

    throw new InvalidArgumentException('Unsupported relation `'.$relationshipName.' of ` type `'.get_class($relation).'`.');
}
```

Reply:
Thanks @nuernbergerA. Done.